Tests/BasicsTests/FileSystem/PathTests.swift

```diff
@@ -6,342 +6,364 @@
-        XCTAssertEqual(RelativePath("../a.b.c").basenameWithoutExt, "a.b")
-        XCTAssertEqual(RelativePath("abc.xyz.123").basenameWithoutExt, "abc.xyz")
+    @Test
+    func baseNameWithoutExt() {
+        #expect(AbsolutePath("/").basenameWithoutExt == (windows ? #"\"# : "/"))
+        #expect(AbsolutePath("/a").basenameWithoutExt == "a")
+        #expect(AbsolutePath("/./a").basenameWithoutExt == "a")
+        #expect(AbsolutePath("/../..").basenameWithoutExt == "/")
+        #expect(RelativePath("../..").basenameWithoutExt == "..")
+        #expect(RelativePath("../a").basenameWithoutExt == "a")
+        #expect(RelativePath("../a/..").basenameWithoutExt == "..")
+        #expect(RelativePath("a/..").basenameWithoutExt == ".")
+        #expect(RelativePath("./..").basenameWithoutExt == "..")
+        #expect(RelativePath("a/../////../////./////").basenameWithoutExt == "..")
+        #expect(RelativePath("abc").basenameWithoutExt == "abc")
+        #expect(RelativePath("").basenameWithoutExt == ".")
+        #expect(RelativePath(".").basenameWithoutExt == ".")
+
+        #expect(AbsolutePath("/a.txt").basenameWithoutExt == "a")
+        #expect(AbsolutePath("/./a.txt").basenameWithoutExt == "a")
+        #expect(RelativePath("../a.bc").basenameWithoutExt == "a")
+        #expect(RelativePath("abc.swift").basenameWithoutExt == "abc")
+        #expect(RelativePath("../a.b.c").basenameWithoutExt == "a.b")
+        #expect(RelativePath("abc.xyz.123").basenameWithoutExt == "abc.xyz")
     }
 
-    func testSuffixExtraction() {
-        XCTAssertEqual(RelativePath("a").suffix, nil)
-        XCTAssertEqual(RelativePath("a").extension, nil)
-        XCTAssertEqual(RelativePath("a.").suffix, nil)
-        XCTAssertEqual(RelativePath("a.").extension, nil)
-        XCTAssertEqual(RelativePath(".a").suffix, nil)
-        XCTAssertEqual(RelativePath(".a").extension, nil)
-        XCTAssertEqual(RelativePath("").suffix, nil)
-        XCTAssertEqual(RelativePath("").extension, nil)
-        XCTAssertEqual(RelativePath(".").suffix, nil)
-        XCTAssertEqual(RelativePath(".").extension, nil)
-        XCTAssertEqual(RelativePath("..").suffix, nil)
-        XCTAssertEqual(RelativePath("..").extension, nil)
-        XCTAssertEqual(RelativePath("a.foo").suffix, ".foo")
-        XCTAssertEqual(RelativePath("a.foo").extension, "foo")
-        XCTAssertEqual(RelativePath(".a.foo").suffix, ".foo")
-        XCTAssertEqual(RelativePath(".a.foo").extension, "foo")
-        XCTAssertEqual(RelativePath(".a.foo.bar").suffix, ".bar")
-        XCTAssertEqual(RelativePath(".a.foo.bar").extension, "bar")
-        XCTAssertEqual(RelativePath("a.foo.bar").suffix, ".bar")
-        XCTAssertEqual(RelativePath("a.foo.bar").extension, "bar")
-        XCTAssertEqual(RelativePath(".a.foo.bar.baz").suffix, ".baz")
-        XCTAssertEqual(RelativePath(".a.foo.bar.baz").extension, "baz")
+    @Test
+    func suffixExtraction() {
```

Comment:
The tests in this file look like they could be parameterized.

Reply:
File has been parameterized!